### DISCUSSION TITLE
How can I configure `indent-blankline.nvim` correctly?

### DISCUSSION THREAD

**Discussion by neuromaancer** [LazyVim/LazyVim#1349] [Q&A] [answered]
I want to personalize the `indent-blankline.nvim` by overriding the default setting in LazyVim. I created a file in plugin folder named `indent-blankline.lua`, and put:
```lua
vim.opt.list = true
vim.cmd([[highlight IndentBlanklineIndent1 guifg=#E06C75 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent2 guifg=#E5C07B gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent3 guifg=#98C379 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent4 guifg=#56B6C2 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent5 guifg=#61AFEF gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent6 guifg=#C678DD gui=nocombine]])

return {
  "lukas-reineke/indent-blankline.nvim",
  event = { "BufReadPost", "BufNewFile" },
  --[[ show_end_of_line = true, ]]
  space_char_blankline = " ",
  show_current_context = true,
  show_current_context_start = true,
  char_highlight_list = {
    "IndentBlanklineIndent1",
    "IndentBlanklineIndent2",
    "IndentBlanklineIndent3",
    "IndentBlanklineIndent4",
    "IndentBlanklineIndent5",
    "IndentBlanklineIndent6",
  },
}
```
This is actually the example in [indent-blankline.nvim](https://github.com/lukas-reineke/indent-blankline.nvim), but it doesn't work like what I expected. Does anyone know how to do it?

**Comment by dpetka2001** [comment, marked as answer]
> vim.opt.list = true
vim.cmd([[highlight IndentBlanklineIndent1 guifg=#E06C75 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent2 guifg=#E5C07B gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent3 guifg=#98C379 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent4 guifg=#56B6C2 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent5 guifg=#61AFEF gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent6 guifg=#C678DD gui=nocombine]])

Put these in `~/.config/nvim/lua/config/options.lua` and the following in `~/.config/nvim/lua/plugins/indent-blankline.lua`
```lua
return {
	{
		"lukas-reineke/indent-blankline.nvim",
		opts = {
			space_char_blankline = " ",
			show_current_context = true,
			show_current_context_start = true,
			char_highlight_list = {
				"IndentBlanklineIndent1",
				"IndentBlanklineIndent2",
				"IndentBlanklineIndent3",
				"IndentBlanklineIndent4",
				"IndentBlanklineIndent5",
				"IndentBlanklineIndent6",
			},
		},
	},
}
```
Test if it works for you.

**Reply by neuromaancer** [reply]
Thank you! it works!

**Reply by davidgao7** [reply]
Hi there @dpetka2001 , thank you for your provided solution;

I was getting `.../share/nvim/lazy/indent-blankline.nvim/lua/ibl/utils.lua:19: 'show_current_context_start' is not a valid key of ibl.config
`

---
I put this in lua/plugins/
indent-blankline.lua
```lua
return {
  {
    "lukas-reineke/indent-blankline.nvim",
    opts = {
      space_char_blankline = " ",
      show_current_context = true,
      show_current_context_start = true,
      char_highlight_list = {
        -- these color can be find in options.lua, you can customize it
        "IndentBlanklineIndent1",
        "IndentBlanklineIndent2",
        "IndentBlanklineIndent3",
        "IndentBlanklineIndent4",
        "IndentBlanklineIndent5",
        "IndentBlanklineIndent6",
      },
    },
  },
}
```

and add these lines to 
option.lua
```lua
-- indent blankline configuration
vim.opt.list = true
vim.cmd([[highlight IndentBlanklineIndent1 guifg=#E06C75 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent2 guifg=#E5C07B gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent3 guifg=#98C379 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent4 guifg=#56B6C2 gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent5 guifg=#61AFEF gui=nocombine]])
vim.cmd([[highlight IndentBlanklineIndent6 guifg=#C678DD gui=nocombine]])
```
---

this is the stacktrace
<img width="783" alt="Screenshot 2024-03-28 at 12 18 20 PM" src="https://github.com/LazyVim/LazyVim/assets/39502459/8afa6be2-da3e-4ce9-a726-e81195816022">

**Reply by dpetka2001** [reply]
That solution is for the old version 2 of `indent-blankline`. You should read the docs of the plugin for the new settings that it has (since version 3 onwards).
